### PR TITLE
Feature/tsjr 313 generate tc user id for existing job candidates

### PR DESCRIPTION
**Pull request by Schpotsky**
Kindly note that this branch is based off [tsjr-297-git-rid-of-ubahn](https://github.com/topcoder-platform/taas-apis/tree/feature/tsjr-297-git-rid-of-ubahn) for which a PR was raised at https://github.com/topcoder-platform/taas-apis/pull/709

The updates of TSJR-313 are in this commit https://github.com/topcoder-platform/taas-apis/pull/710/commits/8945ce7c6f4c1598683ab3caa9a023bf8df39427

### CODE REVIEW

File: Remove_Ubahn-API_dependency.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Objectives:
```
Comment (testflyjets) [inline Remove_Ubahn-API_dependency.md:1]: Really excellent documentation!

File: scripts/job-candidate/generate-tcUserId-for-existing-candidates.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
+const Sequelize = require('sequelize')
+const _ = require('lodash')
+const models = require('../../src/models')
+const helper = require('../../src/common/helper')
+const logger = require('../../src/common/logger')
+const config = require('config')
+
+// The number of job candidates to process in a single batch
+const JOB_CANDIDATES_BATCH_SIZE = Number(process.env.JOB_CANDIDATES_BATCH_SIZE) || 200
```
Comment (testflyjets) [inline scripts/job-candidate/generate-tcUserId-for-existing-candidates.js:9]: How many total candidates are there, and how long will they take to process with a 1 second delay between batches?

File: scripts/job-candidate/generate-tcUserId-for-existing-candidates.js
Comment (testflyjets) [inline scripts/job-candidate/generate-tcUserId-for-existing-candidates.js:1]: Really clean, well-documented code. Excellent!

File: src/common/helper.js [hunk 3]
```diff
@@ -2287,6 +2190,31 @@ function verifyZoomLinkToken (token) {
   }
 }
 
+/**
+ * Check if the user id (Number) exists in member-api
+ * If it exists return it, otherwise throw en error
+ * @param {number} userId - The Topcoder userId to check
+ */
+async function ensureTopcoderUserIdExists (userId) {
```
Comment (testflyjets) [inline src/common/helper.js:2198]: Any concerns with API throttling if this is called 200 times in a short period of time?

Comment (testflyjets) [reply src/common/helper.js:2198]: Actually, it doesn't appear to be used in the script, so you can probably disregard my question here.

File: scripts/job-candidate/generate-tcUserId-for-existing-candidates.js (hunk 2 above)
Comment (Schpotsky) [reply scripts/job-candidate/generate-tcUserId-for-existing-candidates.js]: @testflyjets ,
We have 24311 job candidates records in Prouduction (8343 unique users - a single user can have multiple job candidates records)

To make the script run faster:
- Raised the batch size to 500
- Removed the wait time between batches
- Provided a static mapping file between the user handles which are present in TaaSDB and userIds for both Production and Dev, calls to member-api to get the member userId by handle will only be done for members who are not in the static file.

Details in this commit: https://github.com/topcoder-platform/taas-apis/pull/710/commits/3adc567a13a1acc346519352c1cdc230273ca5f3

**Review by testflyjets — APPROVED** [review]

**Pull request merged**